pkg/mcs/exportspec_test.go

```diff
@@ -40,38 +41,100 @@ func TestMarshalUnmarshal(t *testing.T) {
 		expected *mcs.ExportSpec
 		mutate   func(mcs.ExportSpec) *mcs.ExportSpec
 	}{
-		/*
-			"empty": {},
-			"missing: timestamp": {},
-			"missing: cluster": {},
-			"missing: namespace": {},
-			"missing: name": {},
-			"missing: service":       {},
-			"missing: service.type":  {},
-			"missing: service.ports": {},
-			"simple spec": {},
-			"multiple ports":         {},
-		*/
+
+		"empty": {
+			expected: &mcs.ExportSpec{},
+			mutate:   func(mcs.ExportSpec) *mcs.ExportSpec { return &mcs.ExportSpec{} },
+		},
+
+		"missing: timestamp": {
+			mutate: func(es mcs.ExportSpec) *mcs.ExportSpec {
+				es.CreatedAt = metav1.Unix(0, 0)
+				return &es
+			},
+		},
+
+		"missing: cluster": {
+			mutate: func(es mcs.ExportSpec) *mcs.ExportSpec {
+				es.ClusterID = ""
+				return &es
+			},
+		},
+
+		"missing: namespace": {
+			mutate: func(es mcs.ExportSpec) *mcs.ExportSpec {
+				es.Namespace = ""
+				return &es
+			},
+		},
+
+		"missing: name": {
+			mutate: func(es mcs.ExportSpec) *mcs.ExportSpec {
+				es.Name = ""
+				return &es
+			},
+		},
+
+		"missing: service": {
+			mutate: func(es mcs.ExportSpec) *mcs.ExportSpec {
+				es.Service = mcs.GlobalProperties{}
+				return &es
+			},
+		},
+
+		"missing: service.type": {
+			mutate: func(es mcs.ExportSpec) *mcs.ExportSpec {
+				es.Service.Type = ""
+				return &es
+			},
+		},
+
+		"missing: service.ports": {
+			mutate: func(es mcs.ExportSpec) *mcs.ExportSpec {
+				es.Service.Ports = []mcsv1a1.ServicePort{}
+				return &es
+			},
+		},
+
+		"simple spec": {
+			mutate: func(es mcs.ExportSpec) *mcs.ExportSpec {
+				return &es
+			},
+		},
+
+		"multiple ports": {
+			mutate: func(es mcs.ExportSpec) *mcs.ExportSpec {
+				new_port_81 := mcsv1a1.ServicePort{Port: 81, Name: "http", Protocol: corev1.ProtocolTCP}
```

Comment:
according to the k8s Service spec, ports should have different names.

https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.22/#serviceport-v1-core:
```The name of this port within the service. This must be a DNS_LABEL. All ports within a ServiceSpec must have unique names. When considering the endpoints for a Service, this must match the 'name' field in the EndpointPort. Optional if only one ServicePort is defined on this service.```

You can just rename to "http2", "http3", etc,

Reply:
I Added the required changes